lib/std/tasks.nim

```diff
@@ -0,0 +1,480 @@
+import std/[macros]
+
+
+type
+  Task* = object ## `Task` contains the callback and its arguments.
+    callback: proc (args: pointer) {.nimcall.}
+    args: pointer
```

Comment:
Args must point to the heap and Task needs a destructor so that the heap based storage gets freed reliably.

Reply:
yeah, will do

Reply:
fixed